src/communication/channel.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "channel.hpp"
+#include "log.hpp"
+#include "utils/aosheader.hpp"
+#include "utils/checksum.hpp"
+
+namespace aos::zephyr::communication {
+
+Mutex Channel::sWriteMutex;
+
+Channel::Channel(TransportItf* transport, int port)
+    : mTransport(transport)
+    , mPort(port)
+{
+}
+
+Error Channel::Connect()
+{
+    UniqueLock lock(mMutex);
+
+    LOG_DBG() << "Connect channel port=" << mPort;
+
+    if (mTransport->IsOpened()) {
```

Comment:
yes, but after that I decided that transport in `Init` method will call Connect itself so the channel only checks if it connects. So as we discussed I will use in here blocking mechanism to wait unit it will connected

Reply:
now used blocking connect when transport would be connected or the channel will be closed